transcoder/docs/playback.md

```diff
@@ -0,0 +1,178 @@
+
+The transcoding service is designed to be highly available. When multiple transcoding service instances are deployed at once and configured properly, users should not notice when at least one service fails. This holds true even when the failed instance(s) were transcoding a video being actively played. This is because Kyoo supports _distributed, parallel transcoding_. The service can be configured so that a minimum number of transcoder instances will transcode the parts of the same video. When multiple instances transcode the same parts of the same video at the same time, only one has to succeed for each segments for transcoding to be successful.
+
+Because no two segments are guaranteed to come from the same transcoder instance, it is critical that all segments are entirely independent of each other, and do not overlap. "Parallel segments", or segments covering the same video and same time range that are produced by different instances, must always start with a [I-frame](https://en.wikipedia.org/wiki/Video_compression_picture_types). The start-finish time interval of parallel segments must also match up exactly, with no extra (or missing) frames. Additionally, for interoperability with direct playback, segments must line up exactly with keyframes in the source video. See [here](https://zoriya.dev/blogs/transcoder/) for more information.
+
+## Playback
+
+### Playlist requests
+```mermaid
+sequenceDiagram
+    participant cvp as Client video player
+    box Transcoder service (1..N)
+        participant api as Web API
+        participant jobs as Job worker
+    end
+    box Backend (HA)
+        participant db as Postgres
+        participant fs as Storage
+    end
+
+    cvp ->> api: Request playlist
+    api ->> db:  Get video metadata (segmentation times)
+    db -->> api: Return result
+    critical Metadata not available
+        api -->> api: Generate metadata
+        api -) db: Cache metadata
+    end
+    api ->> api: Generate playlist
+    api -) db: Create transcoding job for first k segments
+    api ->> cvp: Return video playlist
+```
+
+### Segment requests
+```mermaid
+sequenceDiagram
+    participant cvp as Client video player
+    box Transcoder service (1..N)
+        participant api as Web API
+        participant jobs as Job worker
+    end
+    box Backend (HA)
+        participant db as Postgres
+        participant fs as Storage
+    end
+
+    cvp ->> api: Requests video segment
+    loop Until segment is available
```

Comment:
in practice we'd wait for an event from pg (using `LISTEN`), can't we specify that as an arrow instead of a loop?

Reply:
Yea I should change it to this. I wrote this before adding the job completion notifications